src/BaseComponent.ts

```
@@ -18,21 +18,15 @@
+        (<G>() => G extends Pick<T, Prop> ? 1 : 2) extends (
+            <G>() => G extends Record<Prop, T[Prop]> ? 1 : 2
+            ) ? true : false
         ) extends false
         ? never
-        : Prop;
-    }[keyof T],
-    {
-        [K in keyof T]: T[K] extends Function ? never : K;
-    }[keyof T]
->;
+        : (T[Prop] extends Function | null ? never : Prop);
```

Biome

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)